src/model/schema.js

```diff
@@ -416,6 +416,43 @@ export default class Schema {
 	}
 
 	/**
+	 * Removes disallowed by {@link module:engine/model/schema~Schema schema} attributes from given nodes.
+	 * When {@link module:engine/model/batch~Batch batch} parameter is provided then attributes will be removed
+	 * using that batch, by creating {@link module:engine/model/delta/attributedelta~AttributeDelta attribute deltas}.
+	 * Otherwise, attributes will be removed directly from provided nodes using {@link module:engine/model/node~Node node} API.
+	 *
+	 * @param {Iterable.<module:engine/model/node~Node>} nodes Nodes that will be filtered.
+	 * @param {module:engine/model/schema~SchemaPath} inside Path inside which schema will be checked.
+	 * @param {module:engine/model/batch~Batch} [batch] Batch to which the deltas will be added.
+	 */
+	removeDisallowedAttributes( nodes, inside, batch ) {
+		for ( const node of nodes ) {
+			const name = node.is( 'text' ) ? '$text' : node.name;
+			const attributes = Array.from( node.getAttributeKeys() );
+			const queryPath = Schema._normalizeQueryPath( inside );
+
+			// When node with attributes is not allowed in current position.
+			if ( !this.check( { name, attributes, inside: queryPath } ) ) {
+				// Let's remove attributes one by one.
+				// TODO: this should be improved to check all combination of attributes.
+				for ( const attribute of node.getAttributeKeys() ) {
+					if ( !this.check( { name, attributes: attribute, inside: queryPath } ) ) {
+						if ( batch ) {
```

Comment:
This is... sad :D Do I understand correctly that this will be simplified once we got the unified API?

Reply:
Well, yes.

Reply:
I hope so :)